src/web/client/utility/schemaHelper.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { portal_schema_V1, portal_schema_V2 } from "../common/portalSchema";
+import * as Constants from "../common/constants";
+import PowerPlatformExtensionContextManager from "../common/localStore";
+
+export function getPortalSchema(schema: string) {
+    if (schema === portal_schema_V2.entities.dataSourceProperties.schema) {
+        return portal_schema_V2;
+    }
+    return portal_schema_V1;
+}
+
+export function getEntity(entity: string) {
+    const powerPlatformExtensionContext = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext();
+    if (powerPlatformExtensionContext.queryParamsMap.get(Constants.schemaKey.SCHEMA_VERSION) as string === portal_schema_V2.entities.dataSourceProperties.schema) {
+        return powerPlatformExtensionContext.entitiesSchemaMap.get(entity);
+    }
+
+    return powerPlatformExtensionContext.entitiesSchemaMap.get(entity);
+}
+
+export function getAttributeParts(attribute: string): { source: string, relativePath: string } {
+    const attributePathArray = attribute.split('.', 2);
+
+    return { source: attributePathArray[0], relativePath: attributePathArray[1] ?? '' };
+}
+
+export function useBase64Decoding(entity: string, attributeType: string): boolean {
+    return entity === Constants.schemaEntityName.WEBFILES &&
+        (attributeType === Constants.entityAttributesWithBase64Encoding.documentbody || attributeType === Constants.entityAttributesWithBase64Encoding.filecontent);
+}
+
+export function useBase64Encoding(entity: string, attributeType: string): boolean {
+    return entity === Constants.schemaEntityName.WEBFILES &&
```

Comment:
nit: Function names can be isBase64Decoding or shouldBase64Decode, 'use' doesn't convey it's a boolean return type

Reply:
Agree. Need to revisit this later. Out of scope for current PR.